packages/headless/src/composables/calendar/createCalendar.ts

```diff
@@ -0,0 +1,343 @@
+import {
+  computed,
+  nextTick,
+  ref,
+  toValue,
+  useId,
+  watch,
+  type MaybeRefOrGetter,
+  type Ref,
+} from "vue";
+
+export type OnyxHeadlessCalderProps = {
+  /**
+   * Whether the calendar should be disabled, preventing user interaction.
+   */
+  disabled?: boolean;
+  /**
+   * The first day of the week to be displayed.
+   */
+  weekStartDay?: OnyxWeekDays;
+  /**
+   * The earliest selectable date.
+   */
+  min?: Date;
+  /**
+   * The latest selectable date.
+   */
+  max?: Date;
+  /**
+   * The Date that should initially displayed
+   */
+  initialDate?: Date;
+};
+export type OnyxWeekDays =
+  | "Monday"
+  | "Tuesday"
+  | "Wednesday"
+  | "Thursday"
+  | "Friday"
+  | "Saturday"
+  | "Sunday";
+
+const getDayIndex = (dayName: OnyxWeekDays) => {
+  const days = ["Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday", "Sunday"];
+  return days.indexOf(dayName);
+};
+
+const getNormalizedDayIndex = (date: Date, weekStartDay: OnyxWeekDays) => {
+  const day = date.getDay();
+  const start = getDayIndex(weekStartDay);
+  // normalize: 0 = Sunday, 1 = Monday ... 6 = Saturday
+  const normalizedDay = day === 0 ? 6 : day - 1;
+  return (normalizedDay - start + 7) % 7;
+};
+
+const getMidnightDate = (date: Date): Date => {
+  const newDate = new Date(date);
+  newDate.setHours(0, 0, 0, 0);
+  return newDate;
+};
+
+const initializeDate = (props: OnyxHeadlessCalderProps) => {
+  const min = props.min ? getMidnightDate(new Date(props.min)) : null;
+  const max = props.max ? getMidnightDate(new Date(props.max)) : null;
+  const today = getMidnightDate(new Date());
+
+  let initialDate = props.initialDate ? getMidnightDate(new Date(props.initialDate)) : today;
+
+  if (min && initialDate < min) {
+    initialDate = min;
+  }
+  if (max && initialDate > max) {
+    initialDate = max;
+  }
+
+  return initialDate;
+};
+
+export function createCalendar(
+  props: OnyxHeadlessCalderProps & {
+    dayNames: MaybeRefOrGetter<string[]>;
+    buttonRefs: Ref<Record<string, HTMLElement>>;
```

Comment:
We can use the Intl.DateTimeFormat API here, too:

See https://stackoverflow.com/q/30437134
```suggestion
    locale: MaybeRef<string>;
    buttonRefs: Ref<Record<string, HTMLElement>>;
```